watertap/unit_models/electrodialysis_1D.py

```diff
@@ -778,6 +1070,497 @@ def eq_recovery_mass_H2O(self, t):
+            if (
+                not self.config.has_Nernst_diffusion_layer
+            ) and x == self.diluate.length_domain.first():
+                self.potential_nonohm_membrane_x[mem, t, x].fix(0)
+                return Constraint.Skip
+
+            return self.potential_nonohm_membrane_x[mem, t, x] == (
+                Constants.gas_constant
+                * self.diluate.properties[t, x].temperature
+                / Constants.faraday_constant
+                * (
+                    sum(
+                        self.ion_trans_number_membrane[mem, j]
+                        / self.config.property_package.charge_comp[j]
+                        for j in self.cation_set
+                    )
+                    + sum(
+                        self.ion_trans_number_membrane[mem, j]
+                        / self.config.property_package.charge_comp[j]
+                        for j in self.anion_set
+                    )
+                )
+                * log(
+                    sum(
+                        self.conc_mem_surf_mol_x[mem, "cathode_left", t, x, j]
+                        for j in self.ion_set
+                    )
+                    / sum(
+                        self.conc_mem_surf_mol_x[mem, "anode_right", t, x, j]
+                        for j in self.ion_set
+                    )
+                )
+            )  # Note: the log argument is taken as *total ion concentration* while in the original equation as *electrolyte concentration*
+            # This is mathematically valid as the ratio eliminate the difference and electro-neutrality stands.
+
+    def _make_performance_dl_polarization(self):
+
+        self.potential_nonohm_dl_x = Var(
+            self.membrane_set,
+            self.flowsheet().time,
+            self.diluate.length_domain,
+            initialize=0.01,  # to inspect
+            bounds=(-50, 50),
+            units=pyunits.volt,
+            doc="Nonohmic potential in two diffusion layers on the two sides of a membrane",
+        )
+
+        self.potential_ohm_dl_x = Var(
+            self.membrane_set,
+            self.flowsheet().time,
+            self.diluate.length_domain,
+            initialize=0.01,  # to inspect
+            bounds=(0, 50),
+            units=pyunits.volt,
+            doc="Ohmic potential in two diffusion layers on the two sides of a membrane",
+        )
+
+        self.dl_thickness_x = Var(
+            self.membrane_set,
+            self.electrode_side_set,
+            self.flowsheet().time,
+            self.diluate.length_domain,
+            initialize=0.0005,
+            bounds=(0, 1e-2),
+            units=pyunits.m,
+            doc="Thickness of the diffusion layer",
+        )
+
+        @self.Constraint(
+            self.flowsheet().time,
+            self.diluate.length_domain,
+            doc="Calculate length-indexed limitting current density",
+        )
+        def eq_current_dens_lim_x(self, t, x):
+            if (
```

Comment:
Should there be a conditional that raises an exception if anything other that InitialValue is provided?